okhttp/src/main/java/io/grpc/okhttp/Utils.java

```diff
@@ -63,6 +62,28 @@ public static Metadata convertTrailers(List<Header> http2Headers) {
     return TransportFrameUtil.toRawSerializedHeaders(headerValues);
   }
 
+  static ConnectionSpec convertSpec(com.squareup.okhttp.ConnectionSpec spec) {
+    ConnectionSpec.Builder builder = new ConnectionSpec.Builder(spec.isTls());
+    builder.supportsTlsExtensions(spec.supportsTlsExtensions());
+    List<com.squareup.okhttp.TlsVersion> tlsVersionList = spec.tlsVersions();
+    String[] tlsVersions = new String[tlsVersionList.size()];
+    for (int i = 0; i < tlsVersions.length; i++) {
+      tlsVersions[i] = tlsVersionList.get(i).javaName();
+      System.err.println(tlsVersions[i]);
+    }
+    builder.tlsVersions(tlsVersions);
+    List<com.squareup.okhttp.CipherSuite> cipherSuiteList = spec.cipherSuites();
+    CipherSuite[] cipherSuites = new CipherSuite[cipherSuiteList.size()];
+    for (int i = 0; i < cipherSuites.length; i++) {
+      cipherSuites[i] = CipherSuite.valueOf(cipherSuiteList.get(i).name());
+      System.err.println(cipherSuites[i].javaName);
+    }
+    builder.cipherSuites(cipherSuites);
+    return builder.build();
```

Comment:
Do you think it would be more readable to move all builder handling to the very end?

```java
return new ConnectionSpec.Builder(spec.isTls())
    .supportsTlsExtensions(spec.supportsTlsExtensions())
    .tlsVersions(tlsVersions)
    .cipherSuites(cipherSuites)
    .build();
```